src/bentoml/_internal/marshal/dispatcher.py

```diff
@@ -370,7 +375,25 @@ async def controller(self):
                     await asyncio.sleep(self.tick_interval)
                     continue
 
-                n_call_out = min(self.max_batch_size, n)
+                n_call_out = 0
+                batch_size = 0
+                try:
+                    for input_info in self._queue:
+                        if (
+                            batch_size + input_info[1].sample.batch_size
+                            < self.max_batch_size
+                        ):
+                            n_call_out += 1
+                            batch_size += input_info[1].sample.batch_size
+                        else:
+                            break
```

Comment:
I think this is fine, but it would probably be better if we used the batch size information to calculate everything above, especially when doing the logic for determining when to wait and cancel requests.

Reply:
Yeah that's true, could I do that in a separate PR? 

The problem of getting an accurate queue length and the problem of making sure the batches we create respect the `max_batch_size` ended up being fairly independent, so this code to figure out the `n_call_out` per batch would be unaffected by an accurate count of the queue length. 

I'm also not as familiar with the use of the queue length, so didn't want to accidentally break something by changing the length and/or make the fix to the `max_batch_size` problem more complicated

Reply:
That sounds reasonable! Though I will say I think you should be able to reuse the calculated value for queue length, since if it's greater than `max_batch_size` what you calculated here should probably be used anyway everywhere for the calculations.